cosmwasm/contracts/pyth/src/error.rs

```diff
@@ -0,0 +1,41 @@
+use cosmwasm_std::StdError;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ContractError {
+    /// Message sender not permitted to execute this operation
+    #[error("PermissionDenied")]
+    PermissionDenied,
+
+    /// Wrapped asset not found in the registry
+    #[error("PriceFeedNotFound")]
+    PriceFeedNotFound,
+
+    /// Message emitter is not an accepted data source.
+    #[error("InvalidUpdateMessageEmitter")]
+    InvalidUpdateEmitter,
+
+    /// Message payload cannot be deserialized to a batch attestation
+    #[error("InvalidUpdatePayload")]
+    InvalidUpdatePayload,
+
+    /// Data source does not exists error (on removing data source)
+    #[error("DataSourceDoesNotExists")]
+    DataSourceDoesNotExists,
+    
+    /// Data source already exists error (on adding data source)
+    #[error("DataSourceAlreadyExists")]
+    DataSourceAlreadyExists,
+}
+
+impl ContractError {
```

Comment:
This error type is much better than before for sure! Instead than the inherent impl though, we can be more idiomatic by implementing `From` here which is used by the `?` operator via `FromResidual`:

```suggestion
impl From<ContractError> for StdError {
  fn from(other: ContractError) -> StdError {
    …
  }
}
```

This will give us a working `?` throughout, anywhere you were using `std_err()` can now just be `?` and anywhere using `std()` can now just be `into()`.

Reply:
You are right. will fix it.